plugins/template/src/lib/Template.cpp

```diff
@@ -0,0 +1,87 @@
+#ifndef PLUGIN_VERSION
+#define PLUGIN_VERSION "0.0.0"
+#endif
+
+#ifndef BUILD_VERSION
+#define BUILD_VERSION "TestBuild"
+#endif
+
+#include "Filenames.h"
+#include "TemplateCode.h"
+#include <tclap/CmdLine.h>
+#include <vmicore/plugins/PluginInit.h>
+
+VMI_PLUGIN_API_VERSION_INFO
+
+using VmiCore::ActiveProcessInformation;
+using VmiCore::ILogger;
+using VmiCore::Plugin::IPluginConfig;
+using VmiCore::Plugin::PluginInterface;
+
+// This file contains the main interface between the VmiCore and the plugin. Here resides the init-function as well as
+// the callbacks for every event.
+namespace Template
+{
+
+    namespace
+    {
+        std::unique_ptr<ILogger> logger;
+        PluginInterface* pluginInterface;
+        std::shared_ptr<TemplateCode> templateCode;
+    }
+
+    // As an example when a new process is started this callback is called. Check the PluginInterface.h for additional
+    // available events.
+    void processStartCallback(std::shared_ptr<const ActiveProcessInformation> processInformation)
+    {
+        logger->info("Process started", {{"Process", *processInformation->fullName}, {"Pid", processInformation->pid}});
+        templateCode->doStuffWithProcessStart(*processInformation);
+    }
+
+    // This is the init function. It gets called by the VmiCore and is responsible for setting up logging, command line
+    // parameters and other required functionality.
+    extern "C" bool init(PluginInterface* communicator,
+                         std::shared_ptr<IPluginConfig> config, // NOLINT(performance-unnecessary-value-param)
+                         std::vector<std::string> args)
+    {
+        pluginInterface = communicator;
+
+        // Set up command line parameter with TCLAP as needed.
+        TCLAP::CmdLine cmd("Plugin Template for VMICore.", ' ', PLUGIN_VERSION);
+        TCLAP::ValueArg parameter1{"p",
+                                   "parameter1",
+                                   "Placeholder for a parameter",
+                                   false,
+                                   std::string("parameter1 default value"),
+                                   "type description for parameter1",
+                                   cmd};
+        cmd.parse(args);
+
+        logger = pluginInterface->newNamedLogger(TEMPLATE_LOGGER_NAME);
+        logger->bind({{VmiCore::WRITE_TO_FILE_TAG, LOG_FILENAME}});
```

Comment:
I don't think that this has any value to anyone but us. We use this tag only for our specific processing. For anyone else this would be confusing and unnecessarily bloaty. I would remove this together with `Filenames.h`